userspace/libsinsp/user.cpp

```diff
@@ -122,12 +123,20 @@ sinsp_usergroup_manager::sinsp_usergroup_manager(sinsp* inspector)
 	: m_import_users(true)
 	, m_last_flush_time_ns(0)
 	, m_inspector(inspector)
-#if defined(HAVE_PWD_H) || defined(HAVE_GRP_H)
 	, m_host_root(m_inspector->get_host_root())
-	, m_ns_helper(new libsinsp::procfs_utils::ns_helper(m_host_root))
+{
+#if defined(HAVE_PWD_H) || defined(HAVE_GRP_H)
+	m_ns_helper = new libsinsp::procfs_utils::ns_helper(m_host_root);
 #endif
+}
+
+sinsp_usergroup_manager::~sinsp_usergroup_manager()
 {
+#if defined(HAVE_PWD_H) || defined(HAVE_GRP_H)
+	delete(m_ns_helper);
+#endif
 }
+
```

Comment:
This is a memory issue waiting to happen with two coexisting `sinsp_usergroup_manager`s (would just having two `sinsp`s be enough?)

Reply:
Yep i am just trying various things! I should've found the proper solution (i hate c++, sorry).